narwhal/primary/src/block_remover.rs

```
@@ -209,190 +49,78 @@ pub struct BlockRemover {
-                        if result.is_ok() {
-                            self.cleanup_internal_state(certificates, batches_by_worker)
-                                .await
-                                .map_err(|_err| ())
-                        } else {
-                            Ok(())
-                        }
-                    }
-                };
+        // Send delete requests to individual workers.
+        let mut worker_requests = Vec::new();
+        let batches_by_worker =
+            utils::map_certificate_batches_by_worker(found_certificates.as_slice());
+        for (worker_id, batch_ids) in batches_by_worker.iter() {
+            let worker_name = self
+                .worker_cache
+                .load()
+                .worker(&self.name, worker_id)
+                .expect("Worker id not found")
+                .name;
 
-                // if the clean up is successful we want to send the result
-                // whatever this is. Otherwise, we are sending back an error
-                // response.
-                if cleanup_successful.is_ok() {
-                    Self::unreliable_broadcast(senders, result).await;
-                } else {
-                    Self::unreliable_broadcast(
-                        senders,
-                        Err(BlockRemoverError {
-                            ids: block_ids,
-                            error: BlockRemoverErrorKind::Failed,
-                        }),
-                    )
-                    .await;
-                }
-            }
+            debug!(
+                "Sending DeleteBatches request for batch ids {batch_ids:?} to worker {worker_name}"
+            );
+            worker_requests.push(
+                self.worker_network
+                    .delete_batches(worker_name, batch_ids.clone()),
+            );
         }
-    }
+        try_join_all(worker_requests).await?;
```

`try_join_all` fails fast if any future returns error, without waiting for everything else to complete - is this the desired behavior here?

---

(nvm, this actually seem correct)

---

Yes it is, this matches the previous behavior